src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsStoreTests.cs

```diff
@@ -0,0 +1,385 @@
+
+            IExceptionInstance instance1 = instances[0];
+            Assert.NotNull(instance1);
+            Assert.Equal(ExpectedId1, instance1.Id);
+
+            IExceptionInstance instance2 = instances[1];
+            Assert.NotNull(instance2);
+            Assert.Equal(ExpectedId2, instance2.Id);
+
+            IExceptionInstance instance3 = instances[2];
+            Assert.NotNull(instance3);
+            Assert.Equal(ExpectedId3, instance3.Id);
+        }
+
+        /// <summary>
+        /// Validates adding and removing exceptions will invoke the callback
+        /// and the store will contain the remaining exceptions.
+        /// </summary>
+        [Fact]
+        public async Task ExceptionsStore_AddTwoRemoveOne_ContainsOne()
+        {
+            int ExpectedCount = 1;
+            ulong ExpectedId1 = 1;
+            ulong ExpectedId2 = 2;
+
+            // Arrange
+            AggregateCallback aggregateCallback = new();
+            await using ExceptionsStore store = new(new[] { new PassThroughCallbackFactory(aggregateCallback) });
+
+            IExceptionsNameCache cache = CreateCache();
+
+            // Act
+            ThresholdCallback thresholdCallback1 = new(2, 0);
+            using (CallbackRegistration.Register(aggregateCallback, thresholdCallback1))
+            {
+                AddExceptionInstance(store, cache, ExpectedId1);
+                AddExceptionInstance(store, cache, ExpectedId2);
+
+                await thresholdCallback1.WaitForThresholdsAsync(CommonTestTimeouts.GeneralTimeout);
+            }
+
+            ThresholdCallback thresholdCallback2 = new(0, 1);
+            using (CallbackRegistration.Register(aggregateCallback, thresholdCallback2))
+            {
+                RemoveExceptionInstance(store, ExpectedId2);
+
+                await thresholdCallback2.WaitForThresholdsAsync(CommonTestTimeouts.GeneralTimeout);
+            }
+
+            // Assert
+            IReadOnlyList<IExceptionInstance> instances = store.GetSnapshot();
+            Assert.Equal(ExpectedCount, instances.Count);
+
+            IExceptionInstance instance1 = instances[0];
+            Assert.NotNull(instance1);
+            Assert.Equal(ExpectedId1, instance1.Id);
+        }
+
+        /// <summary>
+        /// Validates adding exceptions and removing all exceptions
+        /// will invoke the callback and the store be empty.
+        /// </summary>
+        [Fact]
+        public async Task ExceptionsStore_AddThreeRemoveThree_Empty()
+        {
+            ulong ExpectedId1 = 1;
+            ulong ExpectedId2 = 2;
+            ulong ExpectedId3 = 3;
+
+            // Arrange
+            AggregateCallback aggregateCallback = new();
+            await using ExceptionsStore store = new(new[] { new PassThroughCallbackFactory(aggregateCallback) });
+
+            IExceptionsNameCache cache = CreateCache();
+
+            // Act
+            ThresholdCallback thresholdCallback1 = new(3, 0);
```

Comment:
nit: For readability, consider naming these more descriptively (one adds, the other removes)